### PR TITLE
Apple Config as singular to podcast

### PR DESCRIPTION
**Pull request by kookster**
- [x] #793

### CODE REVIEW

File: db/migrate/20231011213406_remove_apple_excluded_category.rb [hunk 1]
```diff
@@ -14,6 +14,8 @@ def change
       pub.poll!(eps)
 
       Apple::Config.mark_as_delivered!(pub)
+    rescue => e
```
Comment (kookster) [inline db/migrate/20231011213406_remove_apple_excluded_category.rb:17]: This older migration won't run with the new code and older db structure

File: app/models/apple/config.rb [hunk 2]
```diff
@@ -61,19 +60,21 @@ def self.find_or_build_private_feed(podcast)
       )
     end
 
+    # this doesn't seem to be called anywhere?
+    # is this a helper for setup via console, with that stdin.get?
```
Comment (svevang) [inline app/models/apple/config.rb:64]: Yep, this is used to onboard new shows

Eventually, we can retrofit this as a programmatic pathway for connecting feeds to delegated delivery.

**Comment by kookster** [comment]
the changes to the models now only, but I'm missing adding a unique constraint in the db.

File: app/models/apple/config.rb [hunk 3]
```diff
@@ -61,19 +60,20 @@ def self.find_or_build_private_feed(podcast)
       )
     end
 
+    # TODO: this a helper for onboarding via console, retrofit when the UX catches up
     def self.build_apple_config(podcast, key)
-      if has_apple_config?(podcast)
+      if podcast.apple_config
```
Comment (svevang) [inline app/models/apple/config.rb:65]: Do we need this `has_apple_config?` method anymore?

File: app/models/apple/config.rb [hunk 4]
```diff
@@ -103,27 +103,18 @@ def self.setup_delegated_delivery(podcast, key: nil, apple_config: nil, apple_sh
...
-      if podcast.feeds.map(&:apple_configs).flatten.compact.present?
-        errors.add(:podcast, "can only have one apple config")
-      end
-    end
-
     def feed_podcasts_match
       return unless public_feed.present? && private_feed.present?
 
-      if public_feed.podcast != private_feed.podcast
+      if (public_feed.podcast_id != podcast_id) || (private_feed.podcast_id != podcast_id)
```
Comment (svevang) [inline app/models/apple/config.rb:109]: Good to see that we're validating, the podcasts / feeds line up hierarchically under the podcast_id

File: app/models/apple/config.rb [hunk 5]
```diff
@@ -103,27 +103,18 @@ def self.setup_delegated_delivery(podcast, key: nil, apple_config: nil, apple_sh
...
         errors.add(:public_feed, "must belong to the same podcast as the private feed")
       end
     end
 
     def publish_to_apple?
       return false unless key&.valid?
 
-      public_feed.publish_to_apple?(self)
+      public_feed.publish_to_apple?
```
Comment (svevang) [inline app/models/apple/config.rb:117]: So much cleaner!

File: app/models/apple/show.rb [hunk 6]
```diff
@@ -31,6 +31,15 @@ def self.get_show(api, show_id)
       api.unwrap_response(resp)
     end
 
+    def self.for_podcast(podcast)
```
Comment (svevang) [inline app/models/apple/show.rb:34]: Not a big deal,  but the pattern up to now has been to build derived objects _from_ some other object. so `from_podcast(podcast)` here for the signature?

File: db/migrate/20231114155433_add_podcast_apple_config_unique_constraint.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+class AddPodcastAppleConfigUniqueConstraint < ActiveRecord::Migration[7.0]
+  def change
+    add_index :apple_configs, [:podcast_id], unique: true
```
Comment (svevang) [inline db/migrate/20231114155433_add_podcast_apple_config_unique_constraint.rb:3]: And with rails validations on unique ids across public/private feeds then this transitively applies to the feeds as well `{:podcast_id, :public_feed_id}` etc

**Review by svevang — APPROVED** [review]
Looks good!

Just a couple minor nitpicks on method name and unused class method

File: app/models/apple/config.rb (hunk 3 above)
Comment (kookster) [reply app/models/apple/config.rb:65]: nope! will remove

File: app/models/apple/show.rb (hunk 6 above)
Comment (kookster) [reply app/models/apple/show.rb:34]: for sure!

**Pull request merged**